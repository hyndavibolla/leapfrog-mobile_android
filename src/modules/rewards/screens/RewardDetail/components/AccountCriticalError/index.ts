export { default as AccountCriticalError } from './AccountCriticalError';
