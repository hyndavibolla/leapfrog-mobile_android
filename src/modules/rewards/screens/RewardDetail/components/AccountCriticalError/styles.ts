import { css } from '@emotion/native';
import { COLOR, FONT_SIZE } from '_constants';

export const styles = {
  container: css`
    background-color: ${COLOR.LIGHT_GRAY};
    align-items: center;
    justify-content: space-between;
    flex: 1;
    padding: 16px;
  `,
  infoContainer: css`
    align-items: center;
    justify-content: center;
    flex: 1;
  `,
  actionContainer: css`
    width: 100%;
    align-items: center;
  `,
  signOutBtn: css`
    color: ${COLOR.PRIMARY_BLUE};
    font-size: ${FONT_SIZE.SMALL};
    margin: 20px;
  `,
  titleText: css`
    text-align: center;
    margin-top: 16px;
    color: ${COLOR.BLACK};
  `,
  subtitleText: css`
    text-align: center;
    margin-top: 24px;
    color: ${COLOR.DARK_GRAY};
    font-size: ${FONT_SIZE.SMALLER};
  `,
  linkText: css`
    text-align: center;
    margin-top: 4px;
    color: ${COLOR.PRIMARY_BLUE};
    font-size: ${FONT_SIZE.SMALLER};
  `,
  lineStyle: css`
    border-bottom-color: ${COLOR.DARK_GRAY};
  `,
  dividerContainer: css`
    width: 100%;
  `
};
