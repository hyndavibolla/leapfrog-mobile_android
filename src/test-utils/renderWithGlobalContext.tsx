import React, { ReactElement } from 'react';
import { render } from '@testing-library/react-native';

import { getInitialState, GlobalProvider } from '../state-mgmt/GlobalState';
import { Deps, IGlobalState, IReducer } from '../models/general';
import { getMockDeps } from './getMockDeps';
import { getMockReducer } from './getMockReducer';

export const renderWithGlobalContext = (
  children: ReactElement,
  deps: Deps = getMockDeps(),
  initState: IGlobalState = getInitialState(),
  combinedReducers: IReducer<IGlobalState, any> = getMockReducer()
) => {
  const renderAPI = render(
    <GlobalProvider deps={deps} initState={initState} combinedReducers={combinedReducers}>
      {children}
    </GlobalProvider>
  );
  return { ...renderAPI, mockReducer: combinedReducers, deps };
};
