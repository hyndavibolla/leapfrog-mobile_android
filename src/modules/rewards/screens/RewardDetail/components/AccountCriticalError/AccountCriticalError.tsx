import React, { memo, useContext } from 'react';
import { TouchableOpacity, View } from 'react-native';

import AvatarWarn from '_assets/shared/avatarWarn.svg';
import { Divider } from '_views/shared/Divider';
import { Text } from '_views/shared/Text';
import { Title, TitleType } from '_views/shared/Title';
import { useLogout } from '_state_mgmt/auth/hooks';
import { useTestingHelper } from '_utils/useTestingHelper';
import { FONT_FAMILY } from '_constants';
import { GlobalContext } from '_state_mgmt/GlobalState';

import { styles } from './styles';

export interface Props {
  hideSignOut?: boolean;
}

export const AccountCriticalError = ({ hideSignOut = false }: Props) => {
  const { deps } = useContext(GlobalContext);
  const { getTestIdProps } = useTestingHelper('account-critical-error');
  const [onLogout, isLoggingOut] = useLogout();
  const email = 'sywsolutions@syw.com';

  return (
    <View {...getTestIdProps('container')} style={styles.container}>
      <View style={styles.infoContainer}>
        <AvatarWarn />
        <Title type={TitleType.HEADER} style={styles.titleText}>
          Something went wrong with your account.
        </Title>
        <Text style={styles.subtitleText}>To continue using Shop Your Way Â® MAX, please contact us on:</Text>
        <TouchableOpacity
          onPress={() => deps.nativeHelperService.linking.openURL(`mailto:${email}?subject=SYW MAX - Account Support`)}
          {...getTestIdProps('link')}
        >
          <Text font={FONT_FAMILY.BOLD} style={styles.linkText}>
            {email}
          </Text>
        </TouchableOpacity>
        <Text style={styles.subtitleText}>We apologize for any inconvenience you're experiencing.</Text>
      </View>
      {!hideSignOut && (
        <>
          <View style={styles.dividerContainer}>
            <Divider lineStyle={styles.lineStyle} />
          </View>
          <View style={styles.actionContainer} {...getTestIdProps('logout-container')}>
            <TouchableOpacity onPress={onLogout} disabled={isLoggingOut} {...getTestIdProps('logout-btn')}>
              <Text font={FONT_FAMILY.BOLD} style={styles.signOutBtn}>
                Sign out
              </Text>
            </TouchableOpacity>
          </View>
        </>
      )}
    </View>
  );
};

export default memo(AccountCriticalError);
