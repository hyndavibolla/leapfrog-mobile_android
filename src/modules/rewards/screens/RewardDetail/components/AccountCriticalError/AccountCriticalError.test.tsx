import React from 'react';
import { fireEvent } from '@testing-library/react-native';
import { act } from 'react-test-renderer';

import { renderWithGlobalContext } from '_test_utils/renderWithGlobalContext';
import { wait } from '_utils/wait';
import AccountCriticalError, { Props } from './AccountCriticalError';

describe('AccountCriticalError', () => {
  let props: Props;
  beforeEach(() => {
    props = {};
  });

  it('should render', () => {
    const { toJSON } = renderWithGlobalContext(<AccountCriticalError {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render without sign out link', () => {
    props.hideSignOut = true;
    const { toJSON } = renderWithGlobalContext(<AccountCriticalError {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should open a link', async () => {
    const { getByTestId, deps } = renderWithGlobalContext(<AccountCriticalError {...props} />);
    fireEvent.press(getByTestId('account-critical-error-link'));
    expect(deps.nativeHelperService.linking.openURL).toHaveBeenCalled();
    await act(() => wait(0));
  });

  it('should sign out', async () => {
    const { getByTestId, deps } = renderWithGlobalContext(<AccountCriticalError {...props} />);
    fireEvent.press(getByTestId('account-critical-error-logout-btn'));
    expect(deps.logger.debug).toBeCalledWith('useLogout', { soft: false });
    await act(() => wait(0));
  });
});
