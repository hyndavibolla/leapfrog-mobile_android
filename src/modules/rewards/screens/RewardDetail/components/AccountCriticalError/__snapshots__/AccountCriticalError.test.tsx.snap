// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountCriticalError should render 1`] = `
<View
  accessibilityLabel="account-critical-error-container"
  accessible={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5F7FA",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "space-between",
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
    }
  }
  testID="account-critical-error-container"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      }
    }
  >
    AvatarWarn
    <Text
      style={
        Array [
          Object {
            "color": "#262C2F",
            "fontFamily": "SFProDisplay-Bold",
            "fontSize": 20,
            "lineHeight": 30,
          },
          Object {
            "color": "#262C2F",
            "marginTop": 16,
            "textAlign": "center",
          },
        ]
      }
      type="titleHeader"
    >
      Something went wrong with your account.
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#909BA0",
            "fontFamily": "SFProDisplay-Medium",
            "fontSize": 16,
            "lineHeight": 22,
          },
          Object {
            "color": "#909BA0",
            "fontSize": 16,
            "marginTop": 24,
            "textAlign": "center",
          },
        ]
      }
    >
      To continue using Shop Your Way ® MAX, please contact us on:
    </Text>
    <View
      accessibilityLabel="account-critical-error-link"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="account-critical-error-link"
    >
      <Text
        font="SFProDisplay-Bold"
        style={
          Array [
            Object {
              "color": "#909BA0",
              "fontFamily": "SFProDisplay-Bold",
              "fontSize": 16,
              "lineHeight": 22,
            },
            Object {
              "color": "#0066CC",
              "fontSize": 16,
              "marginTop": 4,
              "textAlign": "center",
            },
          ]
        }
      >
        sywsolutions@syw.com
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#909BA0",
            "fontFamily": "SFProDisplay-Medium",
            "fontSize": 16,
            "lineHeight": 22,
          },
          Object {
            "color": "#909BA0",
            "fontSize": 16,
            "marginTop": 24,
            "textAlign": "center",
          },
        ]
      }
    >
      We apologize for any inconvenience you're experiencing.
    </Text>
  </View>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
            "flexDirection": "column",
            "justifyContent": "space-between",
            "paddingVertical": 16,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#EBEDF2",
              "borderBottomWidth": 1,
            },
            Object {
              "borderBottomColor": "#909BA0",
            },
          ]
        }
      />
    </View>
  </View>
  <View
    accessibilityLabel="account-critical-error-logout-container"
    accessible={true}
    style={
      Object {
        "alignItems": "center",
        "width": "100%",
      }
    }
    testID="account-critical-error-logout-container"
  >
    <View
      accessibilityLabel="account-critical-error-logout-btn"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="account-critical-error-logout-btn"
    >
      <Text
        font="SFProDisplay-Bold"
        style={
          Array [
            Object {
              "color": "#909BA0",
              "fontFamily": "SFProDisplay-Bold",
              "fontSize": 16,
              "lineHeight": 22,
            },
            Object {
              "color": "#0066CC",
              "fontSize": 18,
              "marginBottom": 20,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 20,
            },
          ]
        }
      >
        Sign out
      </Text>
    </View>
  </View>
</View>
`;

exports[`AccountCriticalError should render without sign out link 1`] = `
<View
  accessibilityLabel="account-critical-error-container"
  accessible={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5F7FA",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "space-between",
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
    }
  }
  testID="account-critical-error-container"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      }
    }
  >
    AvatarWarn
    <Text
      style={
        Array [
          Object {
            "color": "#262C2F",
            "fontFamily": "SFProDisplay-Bold",
            "fontSize": 20,
            "lineHeight": 30,
          },
          Object {
            "color": "#262C2F",
            "marginTop": 16,
            "textAlign": "center",
          },
        ]
      }
      type="titleHeader"
    >
      Something went wrong with your account.
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#909BA0",
            "fontFamily": "SFProDisplay-Medium",
            "fontSize": 16,
            "lineHeight": 22,
          },
          Object {
            "color": "#909BA0",
            "fontSize": 16,
            "marginTop": 24,
            "textAlign": "center",
          },
        ]
      }
    >
      To continue using Shop Your Way ® MAX, please contact us on:
    </Text>
    <View
      accessibilityLabel="account-critical-error-link"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="account-critical-error-link"
    >
      <Text
        font="SFProDisplay-Bold"
        style={
          Array [
            Object {
              "color": "#909BA0",
              "fontFamily": "SFProDisplay-Bold",
              "fontSize": 16,
              "lineHeight": 22,
            },
            Object {
              "color": "#0066CC",
              "fontSize": 16,
              "marginTop": 4,
              "textAlign": "center",
            },
          ]
        }
      >
        sywsolutions@syw.com
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#909BA0",
            "fontFamily": "SFProDisplay-Medium",
            "fontSize": 16,
            "lineHeight": 22,
          },
          Object {
            "color": "#909BA0",
            "fontSize": 16,
            "marginTop": 24,
            "textAlign": "center",
          },
        ]
      }
    >
      We apologize for any inconvenience you're experiencing.
    </Text>
  </View>
</View>
`;
